H/HDF5/build_tarballs.jl

```diff
@@ -286,12 +298,13 @@ products = [
 # Dependencies that must be installed before this package can be built
 dependencies = [
     # To ensure that the correct version of libgfortran is found at runtime
-    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
-    Dependency("LibCURL_jll"; compat="7.73,8"),
-    Dependency("OpenSSL_jll"; compat="3.0.8"),
-    Dependency("Zlib_jll"),
-    # Dependency("dlfcn_win32_jll"; platforms=filter(Sys.iswindows, platforms)),
-    Dependency("libaec_jll"),   # This is the successor of szlib
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"), v"1.1.1";
+               compat="1.1"),
+    Dependency("LibCURL_jll", v"8.7.1"; compat="7, 8"),
+    Dependency("OpenSSL_jll", v"3.0.13"; compat="3"),
+    Dependency("Zlib_jll", v"1.3.1"; compat="1.2"),
```

Comment:
Again it's counterintuitive to build with a version later the minimum compatible one 
```suggestion
    Dependency("Zlib_jll"; compat="1.2.1"),
```